### ISSUE TITLE
zabbix_mediatype parameter renamed in Zabbix 4.4 and onwards

### ISSUE THREAD

**Issue by sky-joker** [ansible-collections/community.zabbix#33]
_From @atkhan007 on 9 Nov 2019_

##### SUMMARY
Mediatype parameter is renamed from Zabbix 4.4 onwards and it breaks the zabbix_mediatype ansible module.

##### ISSUE TYPE
Zabbix 4.4 renamed the mediatype parameter from 'description' to 'name', hence breaking the module.

##### COMPONENT NAME
zabbix_mediatype.py

##### ANSIBLE VERSION
ansible 2.9.0
config file = /home/user/ansible/ansible.cfg
configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
ansible python module location = /usr/lib/python2.7/dist-packages/ansible
executable location = /usr/bin/ansible
python version = 2.7.16 (default, Oct 7 2019, 17:36:04) [GCC 8.3.0]

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT
Ubuntu 19.04 64-bit 5.0.0-32-generic [#34](https://github.com/ansible/ansible/issues/34)-Ubuntu

##### STEPS TO REPRODUCE

```yaml
---
- name: Create Testing mediatype
  zabbix_mediatype:
  name: "Test script"
  server_url: "https://localhost/zabbix/"
  login_user: "Admin"
  login_password: "zabbix"
  type: 'script'
  script_name: 'test_script.sh'
  validate_certs: no
  state: present
```

##### EXPECTED RESULTS
```
TASK [zabbix_server : Create Testing mediatype] ***************************************************************************
changed: [zabbix-server] => {
    "changed": true, 
    "invocation": {
        "module_args": {
            "attempt_interval": 10, 
            "gsm_modem": null, 
            "http_login_password": null, 
            "http_login_user": null, 
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "login_user": "Admin", 
            "max_attempts": 3, 
            "max_sessions": 1, 
            "message_text_limit": null, 
            "name": "Test script", 
            "password": null, 
            "script_name": "test_script.sh", 
            "script_params": null, 
            "server_url": "https://localhost/********/", 
            "smtp_authentication": false, 
            "smtp_email": null, 
            "smtp_helo": "localhost", 
            "smtp_security": null, 
            "smtp_server": "localhost", 
            "smtp_server_port": 25, 
            "smtp_verify_host": false, 
            "smtp_verify_peer": false, 
            "state": "present", 
            "status": "enabled", 
            "timeout": 10, 
            "type": "script", 
            "username": null, 
            "validate_certs": false
        }
    }, 
    "msg": "Mediatype created: Test script, ID: None"
}
```

##### ACTUAL RESULTS

```paste below
TASK [zabbix_server : Create Testing mediatype] ***************************************************************************
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_********_mediatype_payload_EI1rVw/ansible_********_mediatype_payload.zip/ansible/modules/monitoring/********/********_mediatype.py", line 487, in create_mediatype                                                               
  File "/usr/local/lib/python2.7/dist-packages/********_api.py", line 341, in method
    return self.universal("%s.%s" % (self.data["prefix"], name), opts[0])
  File "/usr/local/lib/python2.7/dist-packages/********_api.py", line 79, in wrapper
    return self.do_request(self.json_obj(method, opts))['result']
  File "/usr/local/lib/python2.7/dist-packages/********_api.py", line 348, in do_request
    return self.parent.do_request(req)
  File "/usr/local/lib/python2.7/dist-packages/********_api.py", line 299, in do_request
    raise ZabbixAPIException(msg, jobj['error']['code'])

fatal: [zabbix-server]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "attempt_interval": 10, 
            "gsm_modem": null, 
            "http_login_password": null, 
            "http_login_user": null, 
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "login_user": "Admin", 
            "max_attempts": 3, 
            "max_sessions": 1, 
            "message_text_limit": null, 
            "name": "Test script", 
            "password": null, 
            "script_name": "test_script.sh", 
            "script_params": null, 
            "server_url": "https://localhost/********/", 
            "smtp_authentication": false, 
            "smtp_email": null, 
            "smtp_helo": "localhost", 
            "smtp_security": null, 
            "smtp_server": "localhost", 
            "smtp_server_port": 25, 
            "smtp_verify_host": false, 
            "smtp_verify_peer": false, 
            "state": "present", 
            "status": "enabled", 
            "timeout": 10, 
            "type": "script", 
            "username": null, 
            "validate_certs": false
        }
    }, 
    "msg": "Failed to create mediatype 'Test script': (u'Error -32602: Invalid params., Media type is missing parameters: name while sending {\"params\": {\"status\": \"0\", \"description\": \"Test script\", \"attempt_interval\": \"10\", \"exec_path\": \"test_script.sh\", \"maxattempts\": \"3\", \"maxsessions\": \"1\", \"exec_params\": \"\", \"type\": \"1\"}, \"jsonrpc\": \"2.0\", \"method\": \"mediatype.create\", \"auth\": \"c1e1525971dcddf5a53f74f4303d8da5\", \"id\": 2}', -32602)"     
}
```

_Copied from original issue: https://github.com/ansible/ansible/issues/64604_

**Comment by D3DeFi** [comment]
Thank you so much for transferring some of the issues from upstream to here @sky-joker ! I've been wanting to do that for the last couple of weeks, but always managed to do something else instead :( 

Can you please close each respective upstream issue (those living in ansible/ansible) that you've opened here since they no longer can be addressed in there?

Except the issues, there are couple of PRs. For example ansible/ansible#64607 that ought to fix this specific issue. Do we ask contributors to reopen PRs here and give them a few days before doing that by ourselves or do we omit the contributors entirely? 

The problem with committing those PRs ourselves is that any original contributor will be lost in the process until we somehow can cherry-pick them across repositories. 

Just for the record, I've counted around 8 issues and 6 PRs still open in the ansible repo regarding zabbix that are more or less still work to be done.

**Comment by sky-joker** [comment]
Sorry for the late reply.  
Thank you for you have closed https://github.com/ansible/ansible/issues/64604.

> Except the issues, there are couple of PRs. For example ansible/ansible#64607 that ought to fix this specific issue. Do we ask contributors to reopen PRs here and give them a few days before doing that by ourselves or do we omit the contributors entirely?

For existing Pull Requests, let's ask the contributor to submit a Pull Request to this repository again.  
If there is no action after waiting for a few weeks (About a month at most?), how about we fix the bug ourselves?

**Comment by D3DeFi** [comment]
Proposed fix in ansible/ansible#64607

**Comment by D3DeFi** [comment]
most of this was fixed by you @sky-joker in ansible/ansible#67693, thus we won't need to port neither ansible/ansible#68333 or ansible/ansible#64604 here.

I have some additional changes for mediatype in mind, so I will double check this and close the issue
